src/components/refactor/dialogs/connectivity/connectivity-form.js

```diff
@@ -0,0 +1,248 @@
+    } = watch(`${CONNECTIVITY}.${VOLTAGE_LEVEL}`) || {};
+
+    useEffect(() => {
+        fetchVoltageLevels(studyUuid, currentNode?.id).then((values) => {
+            setVoltageLevelOptions(
+                values.sort((a, b) => a.id.localeCompare(b.id))
+            );
+        });
+    }, [studyUuid, currentNode?.id]);
+
+    useEffect(() => {
+        if (watchVoltageLevelId) {
+            switch (watchVoltageLevelTopologyKind) {
+                case 'NODE_BREAKER':
+                    fetchBusbarSectionsForVoltageLevel(
+                        studyUuid,
+                        currentNode?.id,
+                        watchVoltageLevelId
+                    ).then((busbarSections) => {
+                        setBusOrBusbarSectionOptions(busbarSections);
+                    });
+                    break;
+
+                case 'BUS_BREAKER':
+                    fetchBusesForVoltageLevel(
+                        studyUuid,
+                        currentNode?.id,
+                        watchVoltageLevelId
+                    ).then((buses) => setBusOrBusbarSectionOptions(buses));
+                    break;
+
+                default:
+                    setBusOrBusbarSectionOptions([]);
+                    break;
+            }
+        } else {
+            setBusOrBusbarSectionOptions([]);
+        }
+    }, [
+        watchVoltageLevelId,
+        watchVoltageLevelTopologyKind,
+        studyUuid,
+        currentNode?.id,
+    ]);
+
+    const areIdsEqual = useCallback((val1, val2) => val1.id === val2.id, []);
+    const getObjectId = useCallback((object) => object.id, []);
+    const newVoltageLevelField = (
+        <AutocompleteInput
+            isOptionEqualToValue={areIdsEqual}
+            allowNewValue
+            name={`${CONNECTIVITY}.${VOLTAGE_LEVEL}`}
+            label="VoltageLevel"
+            options={voltageLevelOptions}
+            getOptionLabel={getObjectId}
+            size={'small'}
+        />
+    );
+
+    const newBusOrBusbarSectionField = (
+        <AutocompleteInput
+            allowNewValue
+            name={`${CONNECTIVITY}.${BUS_OR_BUSBAR_SECTION}`}
+            label="BusBarBus"
+            options={busOrBusbarSectionOptions}
+            getOptionLabel={getObjectId}
+            isOptionEqualToValue={areIdsEqual}
+            size={'small'}
+        />
+    );
+
+    const newConnectionNameField = (
+        <TextInput
+            name={`${CONNECTIVITY}.${CONNECTION_NAME}`}
+            label="ConnectionName"
+        />
+    );
+
+    const newConnectionDirectionField = (
+        <SelectInput
+            name={`${CONNECTIVITY}.${CONNECTION_DIRECTION}`}
+            label="ConnectionDirection"
+            options={CONNECTION_DIRECTIONS}
+            fullWidth
+            size={'small'}
+        />
+    );
+
+    const handleClickOpenDiagramPane = useCallback(() => {
+        setIsDiagramPaneOpen(true);
+    }, [setIsDiagramPaneOpen]);
```

Comment:
i don't think we put setters in dependencies list, but maybe the code is not very homogeneous

Reply:
fixed